### PR TITLE
Render markdown for description in software templates

### PR DESCRIPTION
**Pull request by andys-hoodie**
## Hey, I just made a Pull Request!

For issue https://github.com/backstage/backstage/issues/7151 

> The description field in Software Template should be rendered via Markdown.

With test description
```
  description: |
    Select the type of entity to create:
    - Resource (e.g. SQL Database, S3 Bucket, ...)
    - Component (e.g. backend service, data pipeline)
    - API
    Find more details in the [System Model](https://backstage.io/docs/features/software-catalog/system-model) documentation.
```

### Current Behavior
image

### Updated Behavior
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/mean-panthers-wonder.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder': minor
```
Comment (benjdlambert) [inline .changeset/mean-panthers-wonder.md:2]: This should be a patch release until we hit version 1.x 👍

**Review by benjdlambert — COMMENTED** [review]
Thanks for this! 👍  Just one small nit about the changeset!

**Comment by benjdlambert** [comment]
@freben yes, you're right. Although I do think this is a good addition, that text that is currently in the card would better be placed as `title` and `description` fields under the `parameters` block. Which I believe is currently already markdown.

**Comment by freben** [comment]
Ah yeah, of course! Thanks

File: .changeset/mean-panthers-wonder.md (hunk 1 above)
Comment (andys-hoodie) [reply .changeset/mean-panthers-wonder.md]: Thanks, done!

Comment (andys-hoodie) [reply .changeset/mean-panthers-wonder.md]: Thanks, done

**Review by Rugvip — APPROVED** [review]
Nice, thanks! 👍

**Pull request merged**